src/resolve/resolved-manager.c

```diff
@@ -1060,6 +1060,60 @@ static int manager_ipv6_send(
         return sendmsg_loop(fd, &mh, 0);
 }
 
+int send_dns_notification(Manager* m, DnsAnswer* answer, const char* query_name)
+{
+        _cleanup_(dns_resource_record_unrefp) DnsResourceRecord *canonical = NULL;
+        _cleanup_free_ char *normalized = NULL;
+        DnsResourceRecord *rr;
+        int ifindex, r;
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        Varlink* connection = NULL;
+
+        DNS_ANSWER_FOREACH_IFINDEX(rr, ifindex, answer) {
+                _cleanup_(json_variant_unrefp) JsonVariant *entry = NULL;
+                int family;
+                const void *p;
```

Comment:
This `void*` is ugly. Please just make separate branches that call `json_build()` below. If/when we add support for other types, branching out like that will be necessary anyway.